float32[4] lidar
float32[4] laser